src/plugins/data/common/search/aggs/param_types/field.ts

```diff
@@ -47,13 +47,38 @@ export class FieldParamType extends BaseParamType {
           );
         }
 
-        if (field.scripted) {
+        const validField = this.getAvailableFields(aggConfig).find(
+          (f: any) => f.name === field.name
+        );
+
+        if (field.type === 'unknown' && !validField) {
+          throw new SavedObjectNotFound('index-pattern-field', field.name);
```

Comment:
why do we throw here ?

Reply:
I thought this is good place for throwing error because this method will called after opening visualization for fetching data. It suits to us, because we want to see visualization page so that we can fix it. Do you know more suitable places for throwing this error?

Reply:
ah sorry, i didn't notice this is the write method. makes sense